presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -57,6 +57,7 @@
     private final Optional<Symbol> leftHashSymbol;
     private final Optional<Symbol> rightHashSymbol;
     private final Optional<DistributionType> distributionType;
+    private final Assignments dynamicFilterAssignments;
```

Comment:
I don't think we need `Assignments` here, do we?
I think we just need to store:
```
List<Symbol> dynamicFilterSymbols;
```
Motivation is: once we create `dynamic_filter` expressions for a given join, then we cannot change order of symbols in dynamic filtering tuples.

Reply:
We might want to keep `Assignments` here, although with a slightly different meaning (basically for validating plan correctness). I will write more tomorrow.

Reply:
Let me go through this comment in detail and revert back.